### PR TITLE
Ftr: add extension module

### PR DESCRIPTION
**Pull request by onewe**
- adapt static registry by extension
- adapt nacos  registry by extension

link #180

### CODE REVIEW

File: common/base/src/extension_param.rs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+use crate::{url::UrlParam, StdError};
```
Comment (yang20150702) [inline common/base/src/extension_param.rs:1]: 缺少 License

File: common/base/src/registry_param.rs [hunk 2]
```diff
@@ -0,0 +1,339 @@
+use crate::{url::UrlParam, StdError, Url};
```
Comment (yang20150702) [inline common/base/src/registry_param.rs:1]: 缺少License

**Comment by onewe** [comment]
done

**Comment by yang20150702** [comment]
Good work!

**Pull request merged**